specs/LaunchingRegisteredProtocols.md

```diff
@@ -0,0 +1,318 @@
+  /// will not be raised regardless of this property. 
+  /// The `LaunchingExternalProtocol` event will be raised after the 
+  /// `NavigationStarting` event and before the `NavigationCompleted` event.
+  /// The default settings will also be updated upon navigation to an external
+  /// protocol. 
+  ///
+  /// If the request is made from a trustworthy origin
+  /// (https://w3c.github.io/webappsec-secure-contexts#potentially-trustworthy-origin) 
+  /// a checkmark box will be displayed on the default browser UI that gives the user 
+  /// the option to always allow the external protocol to launch from this origin.
+  /// If the user checks this box, upon the next request from that origin to the 
+  /// protocol, the event will still be raised, but there will be no default dialog shown
+  /// when `Cancel` is `FALSE`.  
+  ///
+  /// If the request is initiated by a cross-origin iframe without a user gesture, 
+  /// the request will be blocked and the `LaunchingExternalProtocol` event will not 
+  /// be raised.
+  /// \snippet SettingsComponent.cpp LaunchingExternalProtocol
+  HRESULT add_LaunchingExternalProtocol(
+      [in] ICoreWebView2LaunchingExternalProtocolEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+
+  /// Remove an event handler previously added with
+  /// `add_LaunchingExternalProtocol`.
+  HRESULT remove_LaunchingExternalProtocol(
+      [in] EventRegistrationToken token);
+  }
+
+/// Receives the `LaunchingExternalProtocol` event.
+[uuid(e5fea648-79c9-47aa-8314-f471fe627649), object, pointer_default(unique)] 
+interface ICoreWebView2LaunchingExternalProtocolEventHandler: IUnknown {
+  /// Provides the event args for the corresponding event.
+  HRESULT Invoke(
+      [in] ICoreWebView2* sender,
+      [in] ICoreWebView2LaunchingExternalProtocolEventArgs* args);
+}
+
+/// Event args for `LaunchingExternalProtocol` event.
+[uuid(fc43b557-9713-4a67-af8d-a76ef3a206e8), object, pointer_default(unique)] 
+interface ICoreWebView2LaunchingExternalProtocolEventArgs: IUnknown {
+  /// The URI with the external protocol to be launched.
+
+  [propget] HRESULT Uri([out, retval] LPWSTR* value);
+
+  /// The origin initiating the external protocol launch.
+  /// The origin will be empty if the WebView2 navigated to the external protocol.
+
+  [propget] HRESULT InitiatingOrigin([out, retval] LPWSTR* value);
+
+  /// `TRUE` when the external protocol request was initiated through a user gesture.
+  ///
+  /// \> [!NOTE]\n\> Being initiated through a user gesture does not mean that user intended
+  /// to access the associated resource.
+
+  [propget] HRESULT IsUserInitiated([out, retval] BOOL* value);
+
+  /// `TRUE` when the external protocol request was initiated via a non-main frame that 
+  /// has a different origin then the owning top-level page. When this property is `TRUE`
+  /// this may indicate a potential security issue and it is advised to block this request.

```

Comment:
Does the system default dialog follow this guidance and block the request, or at least default to No?

Reply:
No, the system default dialog does not follow this guidance and block the request or default to No, it only displays the dialog to the user and allows the user to select to open the external protocol.

Reply:
Doesn't sound like valid advise then, maybe we should remove it here

Reply:
Please weaken or remove the guidance since we don't follow it anywhere.
Also consider do we actually require this property at all?

Reply:
@david-risney I am fine with removing this property, we are not allowing any requests that would be blocked from the default browser.